### PR TITLE
Add intersectsSegmentTriangle method to Vector3

### PR DESCRIPTION
**Pull request by Pirulax**
I was lazy to switch branches lol

This PR exposes the `intersectsSegmentTriangle` method that can be found in the `CVector` class. (its used in testLineAgainstWater)
The performance seems way better(300%, although my algorithm is more optimized, as it has everything possible cached) than the Lua implementation (with vectors. Without it probably would be way better).

Hope this gets merged fast, as the algorithm itself it tested already.

Note:  Haven't looked into the algorithm itself, but im pretty sure its optimized. The bottleneck is our Lua API probably. (Maybe the old parser would be a slightly bit faster)

### CODE REVIEW

**Review by Woovie — APPROVED** [review]

**Comment by Pirulax** [comment]
@sbx320 Could you help me with this? The non msvc builds

**Review by qaisjp — CHANGES_REQUESTED** [review]
1. Non-windows will need fixing
2. Place them underneath "dot", there's no need to start a brand new section (in this case it starts with create/static, then special operations, then getters, then setters, then variables)

**Comment by botder** [comment]
@Pirulax what's the status on this?

**Comment by Pirulax** [comment]
Well should I make it work?

**Comment by botder** [comment]
Sure, why not.

**Comment by patrikjuvonen** [comment]
Hey @Pirulax, what's the status of this pull request, can we expect it for the upcoming release?

**Comment by Pirulax** [comment]
This function can be called without an instance of a vector. Eg.:
```lua
Vector3.intersectsSegmentTriangle(0, 0, 0, 0, 5, 5, 5, 0, 0, 0, 5, 0, 0, 0, 5);
```
```lua
Vector3(0, 0, 0):intersectsSegmentTriangle(0, 5, 5, 5, 0, 0, 0, 5, 0, 0, 0, 5);
```
Will yield the same result.

Should it be added as a global function as well at this point?

**Comment by patrikjuvonen** [comment]
> This function can be called without an instance of a vector. Eg.:
> 
> ```lua
> Vector3.intersectsSegmentTriangle(0, 0, 0, 0, 5, 5, 5, 0, 0, 0, 5, 0, 0, 0, 5);
> ```
> 
> ```lua
> Vector3(0, 0, 0):intersectsSegmentTriangle(0, 5, 5, 5, 0, 0, 0, 5, 0, 0, 0, 5);
> ```
> 
> Will yield the same result.
> 
> Should it be added as a global function as well at this point?

I'm not sure we have other Vector3-based global functions as such so this would stand out unnecessarily. I think it's fine like this.

**Comment by Pirulax** [comment]
Well `getDistanceBetweenPoints` does the same as `(a - b).Length()`
So thats some kind of a wrapper.

File: Shared/mods/deathmatch/logic/lua/LuaBasic.h [hunk 1]
```diff
@@ -11,6 +11,10 @@
 #include <variant>
 #include <array>
 
+#include <CVector2D.h>
+#include <CVector.h>
+#include <CVector4D.h>
+
```
Comment (patrikjuvonen) [inline Shared/mods/deathmatch/logic/lua/LuaBasic.h]: Are these additions needed?

**Comment by Pirulax** [comment]
Just wanna make sure it gets merged to 1.5.9

**Review by patrikjuvonen — APPROVED** [review]
Thanks, tested lightly.

**Comment by patrikjuvonen** [comment]
Please document this change on the Vector3 Wiki page at https://wiki.multitheftauto.com/wiki/Vector/Vector3

**Pull request merged**